lib/waterline/utils/normalize.js

```diff
@@ -50,6 +50,13 @@ var normalize = module.exports = {
       else if (context.attributes[pk].type == 'string') {
         coercePK = function(pk) {return String(pk).toString();};
       }
+      else if (context.attributes[pk].type == 'uuid') {
+        coercePK = function(pk) {
+          var pattern = /[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}/i
+          var matches = pattern.exec(pk);
+          return matches && matches[0];
```

Comment:
I'd rather throw an error, though let's check to make sure the callers wrap this in a try/catch block.